File: Python/SQLAlchemy_ORM/bll/services/order_detail_service.py
from dal.repositories.order_detail_repository import OrderDetailRepository
from dal.data.database import SessionLocal
from dal.entities.order_detail import OrderDetail

class OrderDetailService:
    def __init__(self):
        self.db = SessionLocal()
        self.repo = OrderDetailRepository(self.db)

    def get_by_id(self, order_id: int, product_id: int) -> OrderDetail:
        entity = self.repo.get_by_id(order_id, product_id)
        if not entity:
            raise ValueError(f"Entity {order_id} and {product_id} not found")
        return entity

    def get_all(self):
        return self.repo.get_all()
    
    def create(self, entity: OrderDetail) -> OrderDetail:
        return self.repo.create(entity)

    def delete(self, order_id: int, product_id: int):
        entity = self.repo.delete(order_id, product_id)
        if not entity:
            raise ValueError(f"Entity {order_id} and {product_id} not found")
        return entity

File: Node/Sequelize_ORM/src/bll/services/orderDetailService.js
const OrderDetailRepository = require('../../dal/repositories/orderDetailRepository');

class OrderDetailService {
  async getAll() {
    return await OrderDetailRepository.getAll();
  }

  async getById(orderID, productID) {
    return await OrderDetailRepository.getById(orderID, productID);
  }

  async create(data) {
    return await OrderDetailRepository.create(data);
  }

  async delete(orderID, productID) {
    return await OrderDetailRepository.delete(orderID, productID);
  }
}

module.exports = new OrderDetailService(); 